libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/Ask.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Templates;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Actions
+{
+    public class Ask : SendActivity
+    {
+        [JsonConstructor]
+        public Ask(
+            string text = null,
+            List<string> expectedProperties = null,
+            [CallerFilePath] string callerPath = "",
+            [CallerLineNumber] int callerLine = 0)
+        : base(text, callerPath, callerLine)
+        {
+            this.RegisterSourceLocation(callerPath, callerLine);
```

Comment:
>RegisterSourceLocation [](start = 17, length = 22)

make sure that RegisterSourceLocation isn't called twice, since you are passing to base and registering here.